debian/postinst

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -e
+
+# If the file exists, we understand that meder setup was already run or
+# the device was manually configured at some point. Do nothing.
+if [ ! -f /etc/mender/mender.conf ]; then
+
+    # For non interactive installs, reproduce the default production
+    # configuration with a dummy device type and tenant token.
+    if [ "$DEBIAN_FRONTEND" == "noninteractive" ]; then
+        mender setup \
+            --device-type unknown \
+            --demo=false \
+            --hosted-mender \
+            --tenant-token "Paste your Hosted Mender token here" \
+            --update-poll 1800 \
+            --inventory-poll 28800 \
+            --retry-poll 300
+
+    # Run the interactive setup wizard
+    else
+        mender setup
```

Comment:
I thought there was a standard Debian frontend we should use, and then populate the values above? Did I misunderstand?

Reply:
This might be something not very well communicated by my side. I brought it up in an stand-up and the JIRA ticket is updated.

ref https://tracker.mender.io/browse/MEN-2426

Basically I decided that the `mender setup` wizard is too involved and I didn't want to copy paste all the prompt texts and flow logic into `debconf` requirements. The Debian way is to have templates for all the questions, and then you write a script to handle all the logic (i.e. moving the user from one template to the next). So it is basically copying Alf work and I foresee maintenance burden here.

I have a prototype for this in a branch, so it is still open for discussion.

It might be accepted in Debian, as we are still providing a non interactive mode with defaults. I'll see if I can get some hints on this from Debian community.